redbeat/schedulers.py

```diff
@@ -320,18 +325,22 @@ def schedule(self):
         client = redis(self.app)
 
         with client.pipeline() as pipe:
-            pipe.zrangebyscore(self.app.redbeat_conf.schedule_key, 0, max_due_at)
+            pipe.zrangebyscore(self.app.redbeat_conf.schedule_key, -1, max_due_at, withscores=True)
 
             # peek into the next tick to accuratly calculate sleep between ticks
             pipe.zrangebyscore(self.app.redbeat_conf.schedule_key,
                                '({}'.format(max_due_at),
                                max_due_at + self.max_interval,
-                               start=0, num=1)
+                               start=0, num=1, withscores=True)
             due_tasks, maybe_due = pipe.execute()
 
         logger.info('Loading %d tasks', len(due_tasks) + len(maybe_due))
         d = {}
-        for key in due_tasks + maybe_due:
+        for key, score in due_tasks + maybe_due:
+            if score < 0:
+                logger.info('Scheduler: Removing ended schedule %s', key)
+                client.zrem(self.app.redbeat_conf.schedule_key, key)
```

Comment:
Is there a way we could encode this behaviour in an RRuleEntry?  Perhaps overriding from_key() to raise an Error or something?  I'm hesitant to change the main loop without a deeper review than I'm able to do at the moment.